x-pack/plugins/apm/public/components/shared/search_bar.tsx

```diff
@@ -28,6 +31,42 @@ interface Props {
   kueryBarBoolFilter?: QueryDslQueryContainer[];
 }
 
+function ApmDatePicker() {
+  const { query } = useApmParams('/*');
+
+  if (!('rangeFrom' in query)) {
+    throw new Error('range not available in route parameters');
+  }
+
+  const {
+    rangeFrom,
+    rangeTo,
+    refreshPaused: refreshPausedFromUrl = 'true',
+    refreshInterval: refreshIntervalFromUrl = '0',
+  } = query;
+
+  const refreshPaused = refreshPausedFromUrl
+    ? toBoolean(refreshPausedFromUrl)
+    : undefined;
+  const refreshInterval = refreshIntervalFromUrl
+    ? toNumber(refreshIntervalFromUrl)
+    : undefined;
+
+  const { incrementTimeRangeId } = useTimeRangeId();
+
+  return (
+    <DatePicker
+      rangeFrom={rangeFrom}
+      rangeTo={rangeTo}
+      refreshPaused={refreshPaused}
+      refreshInterval={refreshInterval}
+      onTimeRangeRefresh={() => {
+        incrementTimeRangeId();
```

Comment:
Is it really the responsibility of who calls `DatePicker` to call `incrementTimeRangeId`? I would expect that the DatePicker component to handle it automatically.

Reply:
The DatePicker component itself is being used by the UX app as well, and they're using urlParams still. So we have APMDatePicker and RUMDatePicker that each deal with date picker state differently.